### ISSUE TITLE
`regex_macros` fails to compile

### ISSUE THREAD

**Issue by Ticki** [rust-lang-nursery/regex#134]
```rust
regex-5ba7da685e6a7b39/master/regex_macros/src/lib.rs:532:30: 532:42 error: no method named `parse_expr` found for type `syntax::parse::parser::Parser<'_>` in the current scope
```

**Comment by BurntSushi** [comment]
@Ticki Could you please share the version of `rustc` that you're using? It's working fine for me on:

```
rustc 1.6.0-nightly (9303055f3 2015-11-19)
```

**Comment by Ticki** [comment]
I use `rustc 1.6.0-nightly (1e3e7e73c 2015-11-07)`. I'll try to update rustc and let you know if it fixes it.

**Comment by dginev** [comment]
To add to this issue, using: ```rustc 1.6.0-nightly (1727dee16 2015-11-26)```, I get:

>~/.cargo/registry/src/github.com-88ac128001ac3a9a/regex_macros-0.1.23/src/lib.rs:33:5: 33:28 error: unresolved import `rustc::plugin::Registry`. Could not find `plugin` in `rustc` [E0432]
~/.cargo/registry/src/github.com-88ac128001ac3a9a/regex_macros-0.1.23/src/lib.rs:33 use rustc::plugin::Registry;
>                                                                                                        ^~~~~~~~~~~~~~~~~~~~~~~
>~/.cargo/registry/src/github.com-88ac128001ac3a9a/regex_macros-0.1.23/src/lib.rs:33:5: 33:28 help: run `rustc --explain E0432` to see a detailed explanation
error: aborting due to previous error
Build failed, waiting for other jobs to finish...
Could not compile `regex_macros`.

When compiling regex_macros as a dependency in a cargo-based project. Seems to be a fresh regression, separate from the original report (I had also encountered the parse_expr error, but that was before updating all versions)

**Comment by FredrikNoren** [comment]
:+1: also getting this error. Seems like this might be the culprit: https://github.com/rust-lang/rust/commit/1430a3500076ad504a0b30be77fd2ad4468ea769

**Comment by FredrikNoren** [comment]
This is how it was solved in another lib that uses plugins: https://github.com/kevinmehall/rust-peg/commit/6f4e677a4438fd3796bcac0be7d2aeb187c873dc

**Comment by BurntSushi** [comment]
Fixed by #135
